app/api/services/informationextraction/ixextractors.ts

```diff
@@ -19,6 +19,13 @@ type AllowedPropertyTypes =
     >
   | 'title';
 
+interface ExtractorType extends IXExtractorType {
+  templates: string[];
+}
+interface NewExtractorType extends Omit<IXExtractorType, '_id'> {
+  templates: string[];
+}
```

Comment:
I think this can just be `type NewExtractorType = Omit<ExtractorType, '_id'>;` so you don't repeat the templates definition.

Reply:
That is because we want to avoid inheriting templates as  ObjectIdSchema[], and then having to do .toString() on every access to it.